src/Func.cc

```diff
@@ -845,6 +928,10 @@ function_ingredients::function_ingredients(ScopePtr scope, StmtPtr body)
 		priority = 0;
 
 	this->body = std::move(body);
+
+	// Add this function to whatever module group it was defined in.
+	auto module_group = event_registry->RegisterGroup(EventGroupKind::Module, current_module);
```

Comment:
I'd avoid accessing the global here, and instead pass the current module as an argument into the constructor. Otherwise, this is easy to use this wrongly by constructing a function ingredient when `current_module` isn't set right.